### ISSUE TITLE
Text smoothing is automatically turned on

### ISSUE THREAD

**Issue by denizzzka** [telegramdesktop/tdesktop#27191]
### Steps to reproduce

Just installed (automatically) new Telegram Desktop version

Previous (or maybe 2-3) versions has behaviour as expected

### Expected behaviour

Text renderer should respect system settings. If text smoothing is disabled on whole system it should be disabled in Telegram window too

### Actual behaviour

Text smoothing is enabled

### Operating system

Debian GNU Linux

### Version of Telegram Desktop

4.12.2

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
What do you mean by text smoothing?

**Comment by denizzzka** [comment]
@ilya-fedin anti-aliasing

Briefly, same as described here https://github.com/google/flutter-desktop-embedding/issues/524

**Comment by ilya-fedin** [comment]
@denizzzka please provide a screenshot of the right and bugged behavior

**Comment by denizzzka** [comment]
@ilya-fedin where I can find previous versions of Telegram Desktop?

**Comment by ilya-fedin** [comment]
github releases

**Comment by ilya-fedin** [comment]
@Aokromes I think this is a not a bug or a wontfix

**Comment by denizzzka** [comment]
> I don't see any bug on the second screenshot

On my system anti-aliasing is totally disabled

Screenshot which describes problem more precise:

![image](https://github.com/telegramdesktop/tdesktop/assets/974632/cd3c49f1-4509-4ee7-831a-b7f1c2e0b537)

**Comment by ilya-fedin** [comment]
Sorry but [Qt doesn't support scaling without anti-aliasing](https://github.com/qt/qtbase/blob/2f6fe3a26843ff68c5d3f9af0a2fc3cce6caac22/src/gui/text/unix/qfontconfigdatabase.cpp#L955). Scaling won't be disabled just because you don't like anti-aliasing.

**Comment by denizzzka** [comment]
@ilya-fedin But everything worked in the previous version

**Comment by ilya-fedin** [comment]
This means qt high-dpi scaling wasn't working. There were changes related to that.

**Comment by denizzzka** [comment]
> This means qt high-dpi scaling wasn't working

Can it be disabled manually? I use old 1920x1080 display

**Comment by ilya-fedin** [comment]
I don't think so, sorry. You have to build tdesktop with Qt 5 to be able to do that.

**Comment by denizzzka** [comment]
I still think the problem is in Telegram Desktop, because, Qt docs says:

_High-DPI displays – also known as **retina displays** – are displays with high resolution (pixels) in relation to their physical size (mm), resulting in a high pixel density, or high number of dots per inch (DPI). The increased resolution is used to provide more detailed content on screen (smoother text, more detailed icons), rather than more content (more windows, larger window sizes)._

But I use old 1920x1080 display with 92 DPI

Also I tried to run Qt 6 application and it respects my anti-aliasing settings: (open in new tab)

![image](https://github.com/telegramdesktop/tdesktop/assets/974632/ea2ad9ed-206c-4744-b098-389318ed21af)

**Comment by ilya-fedin** [comment]
I don't think you're right, tdesktop doesn't deal with such low-level stuff.

**Comment by denizzzka** [comment]
I just think the problem might be inside of `isActive()`, which provided by Qt
```c
bool forcedAntialiasSetting = !antialias || QHighDpiScaling::isActive();
```
but can't check that

**Comment by ilya-fedin** [comment]
Apparently. But we can't fix Qt.

**Comment by denizzzka** [comment]
yes, but you distribute BLOB - no one except TG team can check it, at least

**Comment by ilya-fedin** [comment]
If you don't like the blob, you have options for compiling from the source code.

**Comment by denizzzka** [comment]
For further lurkers: fixed in 4.13 version
